### PR TITLE
[CALCITE-5468] SqlToRelConverter throws if ORDER BY contains IN

### PR DESCRIPTION
**Pull request by dssysolyatin**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 1]
```diff
@@ -4769,4 +4769,18 @@ void checkUserDefinedOrderByOver(NullCollation nullCollation) {
...
+   * SqlToRelConverter should register sub-queries inside ORDER BY clause
+   * for queries without aggregation</a>.
+   */
+  @Test void testOrderByWithSubQuery() {
+    String sql = "SELECT empno\n"
+        + "FROM emp\n"
+        + "ORDER BY\n"
+        + "CASE WHEN empno IN (1,2) THEN 0 ELSE 1 END";
+    sql(sql).ok();
+  }
```
Comment (julianhyde) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java:4785]: Don't put stuff on the end. It causes merge conflicts. Put it in the most logical place.

Comment (julianhyde) [reply core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java:4785]: This IN is not a subquery. Can you rename this test, and add a test for a subquery?

Comment (dssysolyatin) [reply core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java]: Yes, IN is not subquery in terms of SQL. But inside codebase there are places with code like:
```
/**
 * Scalar expression that represents an IN, EXISTS or scalar sub-query.
 */
public class RexSubQuery extends RexCall
```
It is actually quite confusing.

I renamed test and added test for subquery.

**Comment by dssysolyatin** [comment]
@julianhyde Can I merge this PR ? Or there is any objections ?

**Comment by julianhyde** [comment]
@dssysolyatin Looks good. Go ahead and merge to main.

**Pull request merged**